packages/flutter/test/material/dropdown_test.dart

```diff
@@ -422,6 +422,67 @@ void main() {
     checkSelectedItemTextGeometry(tester, 'two');
   });
 
+  testWidgets('Dropdown menu scrolls to first item in long lists', (WidgetTester tester) async {
+    final Key buttonKey = UniqueKey();
+    Widget build() => buildFrame(
+      buttonKey: buttonKey,
+      value: null, // nothing selected
+      items: List<String>.generate(/*length=*/ 100, (int index) => index.toString())
+    );
+
+    // Open the dropdown menu
+    await tester.pumpWidget(build());
+    final RenderBox buttonBox = tester.renderObject(find.byKey(buttonKey));
+    assert(buttonBox.attached);
+    await tester.tap(find.byKey(buttonKey));
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 10)); // finish the menu animation
+
+    // Find the first item in the scrollable dropdown list
+    final Finder menuItemFinder = find.byType(Scrollable);
+    final RenderBox menuItemContainer = tester.renderObject<RenderBox>(menuItemFinder);
+    assert(menuItemContainer.attached);
+    final RenderBox firstItem = tester.renderObject<RenderBox>(
+      find.descendant(of: menuItemFinder, matching: find.byKey(const ValueKey<String>('0'))));
+    assert(firstItem.attached);
+
+    // List should be scrolled so that the first item is at the top
+    const Offset selectedItemOffset = Offset(0.0, -8.0);
+    expect(
+      firstItem.size.topCenter(firstItem.localToGlobal(selectedItemOffset)).dy,
+      equals(menuItemContainer.size.topCenter(menuItemContainer.localToGlobal(Offset.zero)).dy)
+    );
+  });
+
+  testWidgets('Dropdown menu aligns selected item with button in long lists', (WidgetTester tester) async {
+    final Key buttonKey = UniqueKey();
+    Widget build() => buildFrame(
+      buttonKey: buttonKey,
+      value: '50',
+      items: List<String>.generate(/*length=*/ 100, (int index) => index.toString())
+    );
+
+    // Open the dropdown menu
+    await tester.pumpWidget(build());
+    final RenderBox buttonBox = tester.renderObject(find.byKey(buttonKey));
+    assert(buttonBox.attached);
+    await tester.tap(find.byKey(buttonKey));
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1)); // finish the menu animation
+
+    // Find the selected item in the scrollable dropdown list
+    final RenderBox selectedItem = tester.renderObject<RenderBox>(
+      find.descendant(of: find.byType(Scrollable), matching: find.byKey(const ValueKey<String>('50'))));
+    assert(selectedItem.attached);
```

Comment:
Why are we asserting `attached` here and elsewhere? I don't believe WidgetTester.renderObject will succeed for an unattached renderer.

... I see, these assertions appear through this file, but not anywhere else (in fact I appear to responsible for a few of them).